### PR TITLE
feat(STONEINTG-402): add integration tests for Sample App

### PR DESCRIPTION
**Pull request by dirgim**
* Add the test_app_endpoint task to find the route and test the endpoint
* Add the integration_test_app pipeline with workspace and defaults
* Add the integrationtestscenario-test-app scenario
* Add the integration_pipelinerun_test_app pipelineRun with workspace secret binding

### CODE REVIEW

**Comment by dheerajodha** [comment]
@dirgim I'm not sure if this is the right way to test this PR but I tried to add this bundle as an Integration Test on RHTAP UI and I'm seeing the below failure on the TaskRun. I think it is due to the face that my namespace doesn't have the needed permissions.

```
STEP-UNNAMED-0

W0427 05:51:39.565535      24 loader.go:223] Config not found: /kubeconfig
Error from server (Forbidden): routes.route.openshift.io "test-app" is forbidden: User "system:serviceaccount:rhn-support-djodha-tenant:pipeline" cannot get resource "routes" in API group "route.openshift.io" in the namespace "default"
Found target host  for app 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: 
{"result":"FAILURE","timestamp":"1682574699","failures":0,"successes":0,"warnings":0}
```

**Comment by MartinBasti** [comment]
My issue is, that pod doesn't even run due mount issues

```
Unable to attach or mount volumes: unmounted volumes=[ws-zdhwp], unattached volumes=[tekton-internal-downward config-service-cabundle-volume tekton-internal-secret-volume-pipeline-dockercfg-g52dh-g6jvr tekton-internal-home tekton-internal-secret-volume-docker-io-pull-k9t82 tekton-internal-run-0 tekton-internal-bin kube-api-access-gxz6b tekton-internal-steps tekton-creds-init-home-0 tekton-internal-results tekton-internal-scripts tekton-internal-workspace config-trusted-cabundle-volume ws-zdhwp tekton-internal-secret-volume-redhat-appstudio-staginguse-f2xhz]: timed out waiting for the condition

MountVolume.SetUp failed for volume "ws-zdhwp" : secret "cluster-credentials-secret" not found
```

**Comment by dirgim** [comment]
Thanks for taking a look @MartinBasti @dheerajodha - I've updated the pipeline definition with a description explaining what the tests expect (App deployed, secrets  mounted, and routes set up).
Unfortunately it's still not possible to fully test this change in end-to-end capacity (either in e2e or locally) because it depends on the entirety of [RHTAP-244](https://issues.redhat.com/browse/RHTAP-244) to be completed, most crucially [ASC-258](https://issues.redhat.com/browse/ASC-258) in order to get the required secrets so the Integration service can mount them when creating pipelineRuns.

File: tasks/test_app_endpoint.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      name: NAMESPACE
+      description: Namespace of the application under test
+    - default: "Hello World"
+      name: EXPECTED_OUTPUT
+      description: Expected output of the application endpoint
+    - default: ""
+      name: PORT
+      description: Application endpoint Port
+  results:
+    - name: HACBS_TEST_OUTPUT
```
Comment (MartinBasti) [inline tasks/test_app_endpoint.yaml]:
```suggestion
    - name: TEST_OUTPUT
```

File: tasks/test_app_endpoint.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ENDPOINT_OUTPUT=$(curl -k https://${HOST}:${PORT})
+        if [[ "${ENDPOINT_OUTPUT}" == "${EXPECTED_OUTPUT}" ]]; then
+          RESULT="SUCCESS"
+        else
+          RESULT="FAILURE"
+        fi
+
+        HACBS_TEST_OUTPUT=$(jq -rc --arg date $(date +%s) --arg RESULT "${RESULT}" --null-input \
+          '{result: $RESULT, timestamp: $date, failures: 0, successes: 0, warnings: 0}')
+        echo -n "${HACBS_TEST_OUTPUT}" | tee $(results.HACBS_TEST_OUTPUT.path)
```
Comment (MartinBasti) [inline tasks/test_app_endpoint.yaml]:
```suggestion
        echo -n "${TEST_OUTPUT}" | tee $(results.TEST_OUTPUT.path)
```

File: tasks/test_app_endpoint.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        
+        # Check the application endpoint
+        ENDPOINT_OUTPUT=$(curl -k https://${HOST}:${PORT})
+        if [[ "${ENDPOINT_OUTPUT}" == "${EXPECTED_OUTPUT}" ]]; then
+          RESULT="SUCCESS"
+        else
+          RESULT="FAILURE"
+        fi
+
+        HACBS_TEST_OUTPUT=$(jq -rc --arg date $(date +%s) --arg RESULT "${RESULT}" --null-input \
```
Comment (MartinBasti) [inline tasks/test_app_endpoint.yaml]:
```suggestion
        TEST_OUTPUT=$(jq -rc --arg date $(date +%s) --arg RESULT "${RESULT}" --null-input \
```

**Review by MartinBasti — COMMENTED** [review]
We should change it to `TEST_OUTPUT`

File: tasks/test_app_endpoint.yaml (hunk 1 above)
Comment (dirgim) [reply tasks/test_app_endpoint.yaml]: Ack, updated

File: tasks/test_app_endpoint.yaml (hunk 2 above)
Comment (dirgim) [reply tasks/test_app_endpoint.yaml]: Ack, updated

File: tasks/test_app_endpoint.yaml (hunk 3 above)
Comment (dirgim) [reply tasks/test_app_endpoint.yaml]: Ack, updated

**Review by jsztuka — APPROVED** [review]
lgtm

**Comment by dirgim** [comment]
After the latest updates in the e2e testing of the [RHTAP-244](https://issues.redhat.com/browse/RHTAP-244) feature, I've confirmed that the example integration test works as intended for the [hacbs-test-project](https://github.com/redhat-appstudio-qe/hacbs-test-project) application.
Example test output:
image

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by MartinBasti — APPROVED** [review]

**Pull request merged**